### PR TITLE
Fix to make sure dbfiles have valid filename

### PR DESCRIPTION
**Pull request by robkooper**
A fix for a critical vulnerability reported by Danny Rosseau at Carve Systems (www.carvesystems.com).

### CODE REVIEW

**Comment by robkooper** [comment]
@gsrohde also update CHANGELOG to capture older commits/pull requests.

File: app/models/db_file.rb [hunk 1]
```diff
@@ -32,6 +32,11 @@ def setup(user_id, upload, args = nil)
     self[:created_user_id] = user_id
     self[:updated_user_id] = user_id
     if upload # Uploaded file
+      # check to see it is a valid filename
+      if upload.original_filename.match(/^[a-zA-Z0-9._\-]+$/).nil?
```
Comment (dlebauer) [inline app/models/db_file.rb]: filenames can only contain letters, numbers, periods and dashes (-)?

File: app/models/db_file.rb [hunk 2]
```diff
@@ -32,6 +32,11 @@ def setup(user_id, upload, args = nil)
...
+        raise("Invalid filename \"#{upload.original_filename}\"")
```
Comment (dlebauer) [inline app/models/db_file.rb]:
```suggestion
        raise("Invalid filename \"#{upload.original_filename}\"; filenames can only contain letters, numbers, periods and dashes.")
```

File: app/models/db_file.rb (hunk 1 above)
Comment (robkooper) [reply app/models/db_file.rb]: correct

Comment (gsrohde) [inline app/models/db_file.rb]: The backslash is extraneous.  But harmless—basically a no-op (I think).

File: app/models/db_file.rb [hunk 3]
```diff
@@ -32,6 +32,11 @@ def setup(user_id, upload, args = nil)
     self[:created_user_id] = user_id
     self[:updated_user_id] = user_id
     if upload # Uploaded file
+      # check to see it is a valid filename
+      if upload.original_filename.match(/^[a-zA-Z0-9._\-]+$/).nil?
+        raise("Invalid filename \"#{upload.original_filename}\"; filenames can only contain letters, numbers, periods and dashes.")
```
Comment (gsrohde) [inline app/models/db_file.rb]: You left out underscores.

**Comment by gsrohde** [comment]
@robkooper This (with the minimal changes I suggest) looks fine to me.  I'm curious, though, to understand the "critical vulnerability" this addresses, information about which I couldn't find on the cited site.

File: app/models/db_file.rb (hunk 3 above)
Comment (dlebauer) [inline app/models/db_file.rb]:
```suggestion
        raise("Invalid filename \"#{upload.original_filename}\"; filenames can only contain letters, numbers, periods, underscores, and dashes.")
```

**Comment by robkooper** [comment]
I have the information about the issue and can share it with you when you get to the office. Until this is released and people have it deployed I'd like to not make it to clear.

**Pull request merged**